crates/cxx-qt-gen/src/generator/naming/property.rs

```diff
@@ -2,32 +2,83 @@
 // SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
-use crate::{naming::Name, parser::property::ParsedQProperty};
+use crate::{
+    naming::Name,
+    parser::property::{FlagState, ParsedQProperty},
+};
 use convert_case::{Case, Casing};
 use quote::format_ident;
 use syn::Ident;
 
+use std::ops::Deref;
+
+#[derive(Debug)]
+pub enum NameState {
+    Auto(Name),
+    Custom(Name),
+}
+
+impl Deref for NameState {
+    type Target = Name;
+
+    fn deref(&self) -> &Self::Target {
+        match self {
+            Self::Auto(name) => name,
+            Self::Custom(name) => name,
+        }
+    }
+}
+
+impl NameState {
+    pub fn from_flag_with_auto_fn(state: &FlagState, auto_fn: impl Fn() -> Name) -> Self {
+        match state {
+            FlagState::Auto => Self::Auto(auto_fn()),
+            FlagState::Custom(ident) => Self::Custom(Name::new(ident.clone())), // TODO: replace this with some sort of type / method name lookup
+        }
+    }
+}
+
 /// Names for parts of a Q_PROPERTY
 pub struct QPropertyNames {
     pub name: Name,
-    pub getter: Name,
+    pub getter: NameState,
     pub getter_wrapper: Name,
-    pub setter: Name,
-    pub setter_wrapper: Name,
-    pub notify: Name,
+    pub setter: Option<NameState>,
+    pub setter_wrapper: Option<Name>,
+    pub notify: Option<NameState>,
 }
 
 impl From<&ParsedQProperty> for QPropertyNames {
     fn from(property: &ParsedQProperty) -> Self {
         let property_name = property_name_from_rust_name(property.ident.clone());
-        let getter = getter_name_from_property(&property_name);
-        let setter = setter_name_from_property(&property_name);
+
+        // Cache flags as they are accessed multiple times
+        let flags = &property.flags;
+
+        let getter = NameState::from_flag_with_auto_fn(&flags.read, || {
+            getter_name_from_property(&property_name)
+        });
+
+        let setter = flags.write.as_ref().map(|setter| {
+            NameState::from_flag_with_auto_fn(setter, || setter_name_from_property(&property_name))
+        });
+
+        let notify = flags.notify.as_ref().map(|notify| {
+            NameState::from_flag_with_auto_fn(notify, || notify_name_from_property(&property_name))
+        });
+
+        let setter_wrapper = if let Some(NameState::Auto(ref setter)) = setter {
+            Some(wrapper_name_from_function_name(setter))
+        } else {
+            None
+        };
+
         Self {
             getter_wrapper: wrapper_name_from_function_name(&getter),
```

Comment:
if the `getter` is Custom here, shouldn't this then be None?

If so then getter_wrapper needs to become `Option<Name>` ? so then some of my previous comments are still valid ?

Reply:
done